src/charm.py

```diff
@@ -90,6 +91,10 @@ def _write_cloud_config(self, data: dict[str, str]) -> None:
         since v2 was removed a long time ago (Queens release)
         https://docs.openstack.org/keystone/latest/contributor/http-api.html
         """
+        os.makedirs(os.path.dirname(OS_CLIENT_CONFIG), exist_ok=True)
```

Comment:
comment: this is harmless, but I think snapd already ensures that the SNAP_COMMON directory exists for every snap

Reply:
I didn't introduce this line; I just moved it up to the beginning of the function block before I created the cacert file.